### PR TITLE
 Notify for availability changes on Android

### PR DESCRIPTION
**Pull request by rohitsangwan01**
@Sunbreak adding for Windows too , 
so changed condition from  ( IOS and Mac ) to (!Platfrom.isLinux)

### CODE REVIEW

File: packages/quick_blue/android/src/main/kotlin/com/example/quick_blue/QuickBluePlugin.kt [hunk 1]
```diff
@@ -29,25 +32,33 @@ class QuickBluePlugin: FlutterPlugin, MethodCallHandler, EventChannel.StreamHand
...
-
     method.setMethodCallHandler(this)
+    eventAvailabilityChange.setStreamHandler(this)
     eventScanResult.setStreamHandler(this)
 
     context = flutterPluginBinding.applicationContext
     mainThreadHandler = Handler(Looper.getMainLooper())
     bluetoothManager = flutterPluginBinding.applicationContext.getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager
+
+    context.applicationContext.registerReceiver(
```
Comment (Sunbreak) [inline packages/quick_blue/android/src/main/kotlin/com/example/quick_blue/QuickBluePlugin.kt:51]: `context.registerReceiver`

File: packages/quick_blue/android/src/main/kotlin/com/example/quick_blue/QuickBluePlugin.kt [hunk 2]
```diff
@@ -185,21 +204,46 @@ class QuickBluePlugin: FlutterPlugin, MethodCallHandler, EventChannel.StreamHand
...
 
   override fun onCancel(args: Any?) {
     val map = args as? Map<String, Any> ?: return
     when (map["name"]) {
+      "availabilityChange" -> availabilityChangeSink = null
       "scanResult" -> scanResultSink = null
     }
   }
 
+  fun parseAvailabilityState(): Int {
```
Comment (Sunbreak) [inline packages/quick_blue/android/src/main/kotlin/com/example/quick_blue/QuickBluePlugin.kt:228]:
```kt
enum class AvailabilityState(val value: Int) {
  unknown(0),
  resetting(1),
  unsupported(2),
  unauthorized(3),
  poweredOff(4),
  poweredOn(5),
}

fun BluetoothManager.getAvailabilityState(): AvailabilityState {
  val state = adapter?.state ?: return AvailabilityState.unsupported
  return when(state) {
    BluetoothAdapter.STATE_OFF -> AvailabilityState.poweredOff
    BluetoothAdapter.STATE_ON -> AvailabilityState.poweredOn
    BluetoothAdapter.STATE_TURNING_ON -> AvailabilityState.resetting
    BluetoothAdapter.STATE_TURNING_OFF -> AvailabilityState.resetting
    else -> AvailabilityState.unknown
  }
}
```

File: packages/quick_blue/example/lib/main.dart [hunk 3]
```diff
@@ -22,11 +22,12 @@ class MyApp extends StatefulWidget {
 class _MyAppState extends State<MyApp> {
   StreamSubscription<BlueScanResult>? _scanResultSubscription;
   StreamSubscription<AvailabilityState>? _availabilitySubscription;
+  bool isAvailabilityChangeStreamSupported = !Platform.isLinux;
```
Comment (Sunbreak) [inline packages/quick_blue/example/lib/main.dart:25]: `Platform.isIOS || Platform.isMacOS || Platform.isAndroid`

**Comment by rohitsangwan01** [comment]
@Sunbreak all Done

**Pull request merged**